### ISSUE TITLE
Do we always need to download snapshot and targets?

### ISSUE THREAD

**Issue by erickt** [theupdateframework/specification#227]
I'm updating rust-tuf to work with TUF-1.0.30. As part of reading through the spec, I that @rdimitrov changed the spec in #209 to allow us to exit the update-cycle early in [update-timestamp 5.4.3.1](https://theupdateframework.github.io/specification/latest/#update-timestamp). Should we apply this same logic to snapshots and targets?

rust-tuf currently implements this, where it will exit early if:

* When updating the snapshot metadata, check if the trusted timestamp role's snapshot version matches the trusted snapshot version, and has the correct hash value, if present.
* When updating the targets metadata, check if the trusted snapshot role's targets version matches the trusted targets version, and has the correct hash value, if present.
* When updating a delegated targets metadata, check if the delegating targets role's version matches the trusted delegated targets version, and has the correct hash value, if present.
* 
Is this optimization allowed? If so, should we describe it in the spec? If not, why is this dangerous?

**Comment by JustinCappos** [comment]
Thanks for the feedback / question!

The timestamp role contains the hash, version, and size of the snapshot.
These should either all match, or all fail together if the information is
mismatched.


>
>    - When updating a delegated targets metadata, check if the delegating
>    targets role's version matches the trusted delegated targets version, and
>    has the correct hash value, if present.
>
>
The snapshot role contains the version of all of the targets metadata files
(top-level or delegated).  So in this case, I think you may mean to say
snapshot file (or I am confused about what you are saying).

**Comment by joshuagl** [comment]
Yes, I would say this optimisation is allowed. If the metadata on-disk matches what is described (version, hashes) by the describing* verified/trusted metadata (timestamp->snapshot, snapshot->targets), we do not need to download the metadata.

python-tuf also implements these optimisations.

* Do we have a good term for this?

**Comment by erickt** [comment]
Since it’s now optional for us to have the hashes of the metadata, do you still think this optimization is safe if we don’t have hashes? The metadata signatures should still protect us from forgery.

**Comment by erickt** [comment]
> The snapshot role contains the version of all of the targets metadata files
(top-level or delegated).  So in this case, I think you may mean to say
snapshot file (or I am confused about what you are saying).

Oops, yeah I meant the snapshot file.

**Comment by JustinCappos** [comment]
Short answer: Yes, it is safe to only have version numbers of targets files
listed in snapshot.

Long answer:
https://www.usenix.org/system/files/conference/atc17/atc17-kuppusamy.pdf

**Comment by erickt** [comment]
FYI I started exploring this in #226.

> I have wondered that too. I think I've seen "signs metadata for" in some TUF diagrams. If we find a good term, we should put it in a glossary.

@joshuagl / @lukpueh - In rust-tuf, we call this `MetadataDescription`, but it always felt a little awkward. Maybe "Descriptor", although that also seems a little vague.
